src/core/analysis-cache.ts

```diff
@@ -111,3 +114,10 @@ function shallowCopyMap<K, V>(from?: Map<K, V>) {
   }
   return new Map();
 }
+
+function shallowCopyAsyncWorkCache<K, V>(from?: AsyncWorkCache<K, V>) {
```

Comment:
This function seems unnecessary, AsyncWorkCache's constructor already does this.